master/src/main/scala/org/apache/celeborn/service/deploy/master/Master.scala

```diff
@@ -538,6 +579,45 @@ private[celeborn] class Master(
     context.reply(HeartbeatFromWorkerResponse(expiredShuffleKeys, registered))
   }
 
+  private def handleWorkerExclude(
+      context: RpcCallContext,
+      host: String,
+      rpcPort: Int,
+      pushPort: Int,
+      fetchPort: Int,
+      replicatePort: Int,
+      excludeAction: String,
+      requestId: String): Unit = {
+    val targetWorker = new WorkerInfo(
+      host,
+      rpcPort,
+      pushPort,
+      fetchPort,
+      replicatePort,
+      new util.HashMap[String, DiskInfo](),
+      JavaUtils.newConcurrentHashMap[UserIdentifier, ResourceConsumption]())
+    val worker: WorkerInfo = workersSnapShot
+      .asScala
+      .find(_ == targetWorker)
+      .orNull
+    if (worker == null) {
+      logWarning(s"Unknown worker $host:$rpcPort:$pushPort:$fetchPort:$replicatePort" +
```

Comment:
if worker ==null  and  context! =null will return context.reply(WorkerExcludeResponse(true))
Is it suitable?

Reply:
@jiaoqingbo, only `context != null` returns `context.reply(WorkerExcludeResponse(true))`.